src/utils/reshapedreflection.fs

```diff
@@ -48,6 +48,10 @@ module internal ReflectionAdapters =
     let isStaticFlag    f    = hasFlag BindingFlags.Static f
     let isInstanceFlag  f    = hasFlag BindingFlags.Instance f
     let isNonPublicFlag f    = hasFlag BindingFlags.NonPublic f
+ 
+    let isOldCorlib =
+        let attributes = typeof<RuntimeTypeHandle>.GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>() |> Seq.toArray
+        if attributes.Length = 0 then false else attributes.[0].Version = "4.6.24410.01"
```

Comment:
let attributes = typeof<RuntimeTypeHandle>.GetTypeInfo().Assembly.GetCustomAttributes<AssemblyFileVersionAttribute>()
    match Seq.tryHead attributes with
    | Some a -> a.Version = "4.6.24410.01"
    | _ -> false

Reply:
also I assume it should be moved down next to usage

Reply:
and the the bool you can remove the temporary bool completely ....